chainer_chemistry/datasets/molnet/molnet.py

```diff
@@ -0,0 +1,176 @@
+
+import numpy as np
+from chainer.dataset import download
+
+from chainer_chemistry.datasets.numpy_tuple_dataset import NumpyTupleDataset
+from chainer_chemistry.dataset.parsers.csv_file_parser import CSVFileParser
+from chainer_chemistry.dataset.preprocessors.atomic_number_preprocessor import AtomicNumberPreprocessor  # NOQA
+from chainer_chemistry.datasets.molnet.molnet_config import molnet_default_config # NOQA
+
+_root = 'pfnet/chainer/molnet'
+
+
+def get_molnet_dataset(dataset_name, preprocessor=None, labels=None,
+                       split='random', frac_train=.8, frac_valid=.1,
+                       frac_test=.1, seed=777, return_smiles=False,
+                       target_index=None):
+    """Downloads, caches and preprocess MoleculeNet dataset.
+
+    Args:
+        dataset_name (str): MoleculeNet dataset name. If you want to know the
+            detail of MoleculeNet, please refer to
+            `official site <http://moleculenet.ai/datasets-1>`_
+        preprocessor (BasePreprocessor): Preprocessor.
+            This sould be chose base on the network to be trained.
+            If it is None, default `AtomicNumberPreprocessor` is used.
+        labels (str or list): List of target labels.
+        return_smiles (bool): If set to ``True``,
+            smiles array is also returned.
+        target_index (list or None): target index list to partially extract
+            dataset. If `None` (default), all examples are parsed.
+    Returns (dict):
+        dictionary that contains dataset that is already splitted into train,
+        valid and test dataset and 1-d numpy array with dtype=object(string)
+        which is a vector of smiles for each example or `None`.
+
+    """
+    assert dataset_name in molnet_default_config
+    dataset_config = molnet_default_config[dataset_name]
+    labels = labels or dataset_config['tasks']
+    if isinstance(labels, str):
+        labels = [labels, ]
+
+    if preprocessor is None:
+        preprocessor = AtomicNumberPreprocessor()
+
+    if dataset_config['task_type'] == 'regression':
+        def postprocess_label(label_list):
+            return np.asarray(label_list, dtype=np.float32)
+    elif dataset_config['task_type'] == 'classification':
+        def postprocess_label(label_list):
+            label_list = np.asarray(label_list)
+            label_list[np.isnan(label_list)] = -1
+            return label_list.astype(np.int32)
+
+    parser = CSVFileParser(preprocessor, labels=labels,
+                           smiles_col=dataset_config['smiles_columns'],
+                           postprocess_label=postprocess_label)
+    if dataset_config['dataset_type'] == 'one_file_csv':
+        result = parser.parse(get_molnet_filepath(dataset_name),
+                              return_smiles=return_smiles,
+                              target_index=target_index)
+        # TODO(motoki): splitting function or class
+        dataset = result['dataset']
+        if split == 'random':
```

Comment:
I feel the option to return 'original' dataset (no random split) is helpful, but we can do in next PR.

Reply:
I agree. I will make an issue.